src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -176,10 +176,15 @@ public DefineFunctionsResult DefineFunctions(string script)
 
             var udfDefinitions = result.UDFs.Select(udf => new UDFDefinition(
                 udf.Ident.ToString(),
-                udf.Body.ToString(),
-                FormulaType.GetFromStringOrNull(udf.ReturnType.ToString()),
+                udf.Body,
+                FormulaType.GetFromStringOrNull(udf.ReturnType.ToString()) ?? FormulaType.Unknown,
                 udf.IsImperative,
-                udf.Args.Select(arg => new NamedFormulaType(arg.VarIdent.ToString(), FormulaType.GetFromStringOrNull(arg.VarType.ToString()))).ToArray())).ToArray();
+                udf.Args.Select(arg =>
+                {
+                    var formulaType = FormulaType.GetFromStringOrNull(arg.VarType.ToString()) ?? FormulaType.Unknown;

```

Comment:
>FormulaType.GetFromStringOrNull(arg.VarType.ToString()) ?? FormulaType.Unknown [](http://example.com/codeflow?start=38&length=78)

nit: Can we put this in a wrapper since we repeat this.
Maybe an extension method on IdentToken?

Then we can  just write:
arg.GetFormulaType();